src/datasources/cache/__tests__/fake.cache.service.ts

```diff
@@ -12,11 +12,18 @@ export class FakeCacheService implements ICacheService {
     this.cache = {};
   }
 
-  delete(cacheDir: CacheDir): Promise<number> {
-    delete this.cache[cacheDir.key];
+  deleteByKey(key: string): Promise<number> {
+    delete this.cache[key];
     return Promise.resolve(1);
   }
 
+  deleteByKeyPattern(pattern: string): Promise<void> {
+    this.cache = Object.entries(this.cache)
```

Comment:
I don't think we should consider immutability at this level (even on the real cache). We can consider the cache to be mutable (there's a slight performance cost if we consider it otherwise).

Something like the following seems clearer what do you think?

```typescript
for (const key in Object.keys(this.cache)) {
  if (patternRegex.test(key)) {
    delete this.cache[key];
  }
}
```

Reply:
Thanks, yes, I think it's better to keep it simple here and consider it mutable. I've changed the code to meet your suggestion, with a subtle change:
```javascript
for (const key in this.cache) {
  if (patternRegex.test(key)) {
    delete this.cache[key];
  }
}
```
Because `for...in` already iterates over the keys of the object, so `Object.keys()` is not necessary in this case (it would be on a `for...of` loop) 🙂